packages/eslint-plugin/tests/rules/naming-convention.test.ts

```diff
@@ -80,6 +80,7 @@ const formatTestNames: Readonly<Record<
 };
 
 const REPLACE_REGEX = /%/g;
+const IGNORED_REGEX = /^.(?!gnored)/; // negative lookahead to not match `[iI]gnored`
```

Comment:
Shouldn't you include the `[iI]` in the regex?

Reply:
Negative lookaheads are a bit of a PITA ... `^((?!^toJSON$).)*$` ... not sure about performance as well.

I wonder if `filter` should also use `{regex: string; match: boolean;}`?

Reply:
yeah it's something I've realised recently with regexes - they're great at matching things, but terrible at _not_ matching things.

I don't know if people will actively want to use filter like that, but it probably makes sense to make it the same style as the `custom` field that I added late.